### PR TITLE
Allow "systemd-nspawn -D / --volatile=yes" to work

### PR DESCRIPTION
**Pull request by poettering**
Hey, this is for you, @larskarlitski @teg!

### CODE REVIEW

**Comment by larskarlitski** [comment]
Thanks :heart:

File: src/shared/machine-image.c [hunk 1]
```diff
@@ -989,28 +989,51 @@ int image_path_lock(const char *path, int operation, LockFile *global, LockFile
...
         if (getenv_bool("SYSTEMD_NSPAWN_LOCK") == 0) {
                 *local = *global = (LockFile) LOCK_FILE_INIT;
                 return 0;
         }
 
-        if (path_equal(path, "/"))
-                return -EBUSY;
+        /* Prohibit taking exclusive locks on the host image. We can't allow this, since we ourselves are
+         * running off it after all, and we don't want any images to manipulate the host image. We make an
+         * exception for shared locks however: we allow those (and make them NOPs since there's not point in
```
Comment (keszybz) [inline src/shared/machine-image.c]: not point → no point
making → taking
questionnable → questionable
migh → might
of the host root → off the host root
on this → with this

File: src/shared/machine-image.c [hunk 2]
```diff
@@ -1024,12 +1047,12 @@ int image_path_lock(const char *path, int operation, LockFile *global, LockFile
                         return -ENOMEM;
         }
 
-        /* For block devices we don't need the "local" lock, as the major/minor lock above should be sufficient, since
-         * block devices are device local anyway. */
-        if (!path_startswith(path, "/dev")) {
+        /* For block devices we don't need the "local" lock, as the major/minor lock above should be
+         * sufficient, since block devices are device local anyway. */
```
Comment (keszybz) [inline src/shared/machine-image.c]: Hmm, "devices are device local" ?

Comment (poettering) [reply src/shared/machine-image.c]: will replace by "host local"

**Comment by poettering** [comment]
force pushed a new version. Only changes are the typos @keszybz found. ptal!

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**